trace-sdk/src/androidTest/java/io/bitrise/trace/data/management/DataManagerInstrumentedTest.java

```diff
@@ -204,16 +209,25 @@ public void reset_shouldNotBeInitialised() {
 
     /**
      * When the {@link DataManager#handleReceivedData(Data)} is called with a valid {@link Data} that will be
-     * converted to a {@link io.opencensus.proto.metrics.v1.Metric} it should not throw an exception.
+     * converted to a {@link io.opencensus.proto.metrics.v1.Metric} it should not throw an exception and be saved
+     * to the database correctly.
      */
     @Test
-    @UiThreadTest
     public void handleReceivedData_shouldHandleMetricWithoutException() {
         DataManager.getInstance(context);
         ApplicationSessionManager.getInstance().startSession();
+        final TraceDatabase traceDatabase = Room.inMemoryDatabaseBuilder(context, TraceDatabase.class).build();
```

Comment:
Did you know there is [https://github.com/bitrise-io/trace-android-sdk/blob/main/trace-sdk/src/androidTest/java/io/bitrise/trace/data/storage/TestDataStorage.java](https://github.com/bitrise-io/trace-android-sdk/blob/main/trace-sdk/src/androidTest/java/io/bitrise/trace/data/storage/TestDataStorage.java)? Or is it deliberate you did not use that?